From 54fe9008193f968bd08ed969e5cfb83e68c902a6 Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Thu, 10 Feb 2022 17:47:27 +0200
Subject: [PATCH 41/43] clk: Move clk_utils to drivers/net/s32

Moved the clk_utils to the drivers/net/s32 folder since
it is used primarily by other drivers in drivers/net.

Removed the s32gen1_clk_funcs.h file, since it only
contained an unused function.

Updated the required makefiles to reflect the new changes.

Issue: ALB-8352
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/Makefile             |  1 -
 arch/arm/cpu/armv8/s32/s32-gen1/Makefile    |  1 -
 arch/arm/cpu/armv8/s32/s32-gen1/soc.c       |  1 -
 board/freescale/s32-gen1/Makefile           |  2 +-
 drivers/clk/Makefile                        |  1 -
 drivers/clk/s32/include/s32gen1_clk_funcs.h | 12 ------------
 drivers/net/Makefile                        |  3 ++-
 drivers/net/pfeng/Makefile                  |  2 +-
 drivers/{clk => net}/s32/Makefile           |  2 +-
 drivers/{clk => net}/s32/clk_utils.c        |  1 -
 drivers/net/s32/s32gen1_clk_utils.h         | 20 ++++++++++++++++++++
 11 files changed, 25 insertions(+), 21 deletions(-)
 delete mode 100644 drivers/clk/s32/include/s32gen1_clk_funcs.h
 rename drivers/{clk => net}/s32/Makefile (78%)
 rename drivers/{clk => net}/s32/clk_utils.c (98%)
 create mode 100644 drivers/net/s32/s32gen1_clk_utils.h

diff --git a/arch/arm/cpu/armv8/s32/Makefile b/arch/arm/cpu/armv8/s32/Makefile
index 181eea017d..e2eea21ac1 100644
--- a/arch/arm/cpu/armv8/s32/Makefile
+++ b/arch/arm/cpu/armv8/s32/Makefile
@@ -13,6 +13,5 @@ obj-y				+= lowlevel.o
 obj-$(CONFIG_GICSUPPORT)	+= gicsupport.o
 obj-$(CONFIG_MP)		+= mp.o
 obj-$(CONFIG_OF_LIBFDT)		+= fdt.o
-ccflags-y			+= -Idrivers/clk/s32/include
 ccflags-y			+= -Idrivers/misc
 ccflags-y			+= -Iboard/freescale/s32-gen1
diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/Makefile b/arch/arm/cpu/armv8/s32/s32-gen1/Makefile
index 269aca92d3..1ebea36c2d 100644
--- a/arch/arm/cpu/armv8/s32/s32-gen1/Makefile
+++ b/arch/arm/cpu/armv8/s32/s32-gen1/Makefile
@@ -18,5 +18,4 @@ endif
 ccflags-y				+= -I./board/freescale/s32-gen1/
 ccflags-y				+= -I./arch/arm/cpu/armv8/s32/s32-gen1/include
 ccflags-y				+= -I./arch/arm/include/asm/arch-s32/s32-gen1/
-ccflags-y				+= -I./drivers/clk/s32/include/
 ccflags-y				+= -I./drivers/misc/
diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/soc.c b/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
index f11fe07907..0ab8c072e7 100644
--- a/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
+++ b/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
@@ -21,7 +21,6 @@
 #include <dm/uclass.h>
 #include <generic-phy.h>
 #endif
-#include <s32gen1_clk_utils.h>
 
 DECLARE_GLOBAL_DATA_PTR;
 
diff --git a/board/freescale/s32-gen1/Makefile b/board/freescale/s32-gen1/Makefile
index 466a1a1be9..1d2b62fbae 100644
--- a/board/freescale/s32-gen1/Makefile
+++ b/board/freescale/s32-gen1/Makefile
@@ -7,7 +7,7 @@
 
 ccflags-$(CONFIG_S32_GEN1) += -Iarch/$(ARCH)/cpu/armv8/s32
 ccflags-$(CONFIG_S32_GEN1) += -Iboard/freescale/s32-gen1
-ccflags-$(CONFIG_S32_GEN1) += -Idrivers/clk/s32/include
+ccflags-$(CONFIG_S32_GEN1) += -Idrivers/net/s32
 ccflags-$(CONFIG_S32_GEN1) += -Idrivers/misc
 obj-y := common.o
 
diff --git a/drivers/clk/Makefile b/drivers/clk/Makefile
index 7c6e025ffb..aead3ae279 100644
--- a/drivers/clk/Makefile
+++ b/drivers/clk/Makefile
@@ -48,4 +48,3 @@ obj-$(CONFIG_STM32H7) += clk_stm32h7.o
 obj-$(CONFIG_CLK_TI_SCI) += clk-ti-sci.o
 obj-$(CONFIG_CLK_VERSAL) += clk_versal.o
 obj-$(CONFIG_CLK_CDCE9XX) += clk-cdce9xx.o
-obj-$(CONFIG_S32_GEN1) += s32/
diff --git a/drivers/clk/s32/include/s32gen1_clk_funcs.h b/drivers/clk/s32/include/s32gen1_clk_funcs.h
deleted file mode 100644
index 83d8e31792..0000000000
--- a/drivers/clk/s32/include/s32gen1_clk_funcs.h
+++ /dev/null
@@ -1,12 +0,0 @@
-/* SPDX-License-Identifier: BSD-3-Clause */
-/*
- * Copyright 2020-2022 NXP
- */
-#ifndef S32GEN1_CLK_FUNCS_H
-#define S32GEN1_CLK_FUNCS_H
-#include <clk-uclass.h>
-#include <inttypes.h>
-
-int s32gen1_set_parent(struct clk *c, struct clk *p);
-
-#endif /* S32GEN1_CLK_FUNCS_H */
diff --git a/drivers/net/Makefile b/drivers/net/Makefile
index ba92e8dc90..bd88c34d49 100644
--- a/drivers/net/Makefile
+++ b/drivers/net/Makefile
@@ -4,7 +4,7 @@
 # Wolfgang Denk, DENX Software Engineering, wd@denx.de.
 
 ccflags-$(CONFIG_DWC_ETH_QOS_S32CC) += -Iarch/$(ARCH)/include/asm/arch-s32/s32-gen1
-ccflags-$(CONFIG_DWC_ETH_QOS_S32CC) += -Idrivers/clk/s32/include
+ccflags-$(CONFIG_DWC_ETH_QOS_S32CC) += -Idrivers/net/s32
 ccflags-$(CONFIG_DWC_ETH_QOS_S32CC) += -Iboard/freescale/s32-gen1
 
 obj-$(CONFIG_ALTERA_TSE) += altera_tse.o
@@ -90,3 +90,4 @@ obj-y += mscc_eswitch/
 obj-$(CONFIG_HIGMACV300_ETH) += higmacv300.o
 obj-$(CONFIG_MDIO_SANDBOX) += mdio_sandbox.o
 obj-$(CONFIG_FSL_ENETC) += fsl_enetc.o fsl_enetc_mdio.o
+obj-$(CONFIG_S32_GEN1) += s32/
diff --git a/drivers/net/pfeng/Makefile b/drivers/net/pfeng/Makefile
index 7c5533915f..02daa8c4e0 100644
--- a/drivers/net/pfeng/Makefile
+++ b/drivers/net/pfeng/Makefile
@@ -29,4 +29,4 @@ ccflags-y += -Idrivers/net/pfeng/regmap
 ccflags-y += -Idrivers/net/pfeng/hw_abi
 ccflags-y += -Wno-date-time
 ccflags-y += -I./arch/arm/include/asm/arch-s32/s32-gen1
-ccflags-y += -I./drivers/clk/s32/include
+ccflags-y += -I./drivers/net/s32
diff --git a/drivers/clk/s32/Makefile b/drivers/net/s32/Makefile
similarity index 78%
rename from drivers/clk/s32/Makefile
rename to drivers/net/s32/Makefile
index 67c1bf83a8..011d0ccbe0 100644
--- a/drivers/clk/s32/Makefile
+++ b/drivers/net/s32/Makefile
@@ -5,4 +5,4 @@
 obj-y += clk_utils.o
 
 ccflags-y += -Iarch/arm/include/asm/arch-s32/s32-gen1
-ccflags-y += -Idrivers/clk/s32/include
+ccflags-y += -Idrivers/net/s32
diff --git a/drivers/clk/s32/clk_utils.c b/drivers/net/s32/clk_utils.c
similarity index 98%
rename from drivers/clk/s32/clk_utils.c
rename to drivers/net/s32/clk_utils.c
index 003473dc38..615f6a11dc 100644
--- a/drivers/clk/s32/clk_utils.c
+++ b/drivers/net/s32/clk_utils.c
@@ -6,7 +6,6 @@
 #include <asm/arch/clock.h>
 #include <dm/uclass.h>
 #include "s32gen1_clk_utils.h"
-#include <s32gen1_clk_funcs.h>
 
 static struct udevice *get_clk_device(void)
 {
diff --git a/drivers/net/s32/s32gen1_clk_utils.h b/drivers/net/s32/s32gen1_clk_utils.h
new file mode 100644
index 0000000000..ea4bc80480
--- /dev/null
+++ b/drivers/net/s32/s32gen1_clk_utils.h
@@ -0,0 +1,20 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright 2020,2022 NXP
+ */
+#ifndef S32GEN1_CLOCK_UTILS_H
+#define S32GEN1_CLOCK_UTILS_H
+
+#include <clk-uclass.h>
+#include <inttypes.h>
+
+int enable_early_clocks(void);
+
+int s32gen1_enable_dev_clk(const char *name, struct udevice *dev);
+ulong s32gen1_set_dev_clk_rate(const char *name,
+			       struct udevice *dev, ulong rate);
+ulong s32gen1_get_dev_clk_rate(const char *name, struct udevice *dev);
+
+int s32gen1_set_parent_clk_id(ulong clk_id, ulong parent_clk_id);
+
+#endif
-- 
2.17.1

